earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -182,18 +341,22 @@ CreateEmptyDocument(const std::string &rootTagname) throw()
                                             0);// document type object (DTD)
     return doc;
    } catch (...) {
-     return 0;
+     notify(NFY_DEBUG, "Error when trying to create DOMDocument.");
+     return nullptr;
    }
 }
 
 namespace {
 bool
 WriteDocumentImpl(DOMDocument *doc, const std::string &filename) throw()
 {
-
-  InitializeXMLLibrary();
+  
+  if (terminateCache) 
+  {
+    while(purgeGuard::instance().isLocked());
+  }
+  purgeGuard::instance().increment();
```

Comment:
There is a race condition here. What happens if the purgeGuard is locked again before write calls increment?